outlook/indico_outlook/plugin.py

```diff
@@ -148,8 +178,73 @@ def outlook():
+
+    def _user_tracks_favorite_categories(self, user):
+        return OutlookPlugin.user_settings.get(user, 'favorite_categories',
+                                               OutlookPlugin.default_user_settings['favorite_categories'])
+
+    def favorite_event_added(self, user, event, **kwargs):
+        if not self._user_tracks_favorite_events(user):
+            return
+        self._record_change(event, user, OutlookAction.add)
+        self.logger.info('Favorite event added: updating %s in %r', user, event)
+
+    def favorite_event_removed(self, user, event, **kwargs):
+        if not self._user_tracks_favorite_events(user):
+            return
+        self._record_change(event, user, OutlookAction.remove)
+        self.logger.info('Favorite event removed: updating %s in %r', user, event)
+
+    def favorite_category_added(self, user, category, **kwargs):
+        if not self._user_tracks_favorite_categories(user):
+            return
+
+        query = (Event.query
+                 .filter(Event.is_visible_in(category.id),
+                         Event.start_dt > now_utc(),
+                         ~Event.is_deleted)
+                 .options(subqueryload('acl_entries'))
+                 .order_by(Event.start_dt, Event.id))
+        events = [e for e in query if e.can_access(user)]
+        for event in events:
+            self._record_change(event, user, OutlookAction.add)
+            self.logger.info('Favorite category added: user %s added event %r', user, event)
+
+        self.logger.info('Favorite category added: updating %s in %r', user, category)
+
+    def favorite_category_removed(self, user, category, **kwargs):
+        if not self._user_tracks_favorite_categories(user):
+            return
+
+        query = (Event.query
+                 .filter(Event.is_visible_in(category.id),
+                         Event.start_dt > now_utc(),
+                         ~Event.is_deleted)
+                 .options(subqueryload('acl_entries'))
+                 .order_by(Event.start_dt, Event.id))
+        events = [e for e in query if e.can_access(user)]
+        for event in events:
+            self._record_change(event, user, OutlookAction.remove)
+            self.logger.info('Favorite category added: user %s added event %r', user, event)
+
+        self.logger.info('Favorite category removed: updating %s in %r', user, category)
+
+    def category_updated(self, category, changes, **kwargs):
+        if not changes.keys() & {'visibility', 'protection_mode'}:
+            return
+
+        # If the visibility changes, we should probably update all events in the categories
+        # that may suddenly become visible/invisible to users. Not quite sure how to handle this yet.
+
     def event_registration_state_changed(self, registration, **kwargs):
-        if not registration.user:
+        if not (registration.user and self._user_tracks_registered_events(registration.user)):
```

Comment:
IMHO this is clearer when reading the code:

```suggestion
        if not registration.user or not self._user_tracks_registered_events(registration.user):
```